karton/core/backend.py

```diff
@@ -251,38 +252,27 @@ def get_all_tasks(self) -> List[Task]:
             if task_data is not None
         ]
 
-    def register_task(self, task: Task) -> None:
+    def register_task(self, task: Task, pipe=None) -> None:
```

Comment:
this may be a bit annoying, but maybe consider declaring a type `RedisLike = Union[StrictRedis, ???Pipe???]` or something, and  using it for typing here and eveywhere

Reply:
:+1: